test/widgets/inbox_test.dart

```diff
@@ -104,6 +125,24 @@ void main() {
     return findRowByLabel(tester, stream.name);
   }
 
+  /// For the given stream ID, find a stream header, either sticky one
+  /// or a normal one
+  Finder findStreamHeader({
+    required WidgetTester tester,
+    required int streamId,
+    required bool isSticky,
+  }) {
+    final stream = store.streams[streamId]!;
+    return find.ancestor(
+      of: find.text(stream.name),
+      matching: find.byWidgetPredicate((widget) =>
+          widget is StreamHeaderItem &&
+          (isSticky
+              ? widget.parentContext != null
+              : widget.parentContext == null)),
```

Comment:
From the user's perspective, there aren't two kinds of header "sticky" and "not sticky" — there's just one header, which has certain behavior about sticking at the top of the viewport when its section is scrolling past.

For these UI tests, in general we aim to write them in a way that reflects the UI as the user sees it, keeping to a minimum the references to our own UI classes and their properties etc. One major reason for that is to avoid making the tests brittle — we want to be able to rearrange and refactor the internals of the code, and if that doesn't change any behavior the user sees then the tests shouldn't care. If we do have to go update a bunch of tests in order to make such a change, that's a drag.

I think it should be possible to write the tests for this new functionality the same way as we've written the other tests in this file, using `findStreamHeaderRow`. If you're running into obstacles doing that, please start a thread in `#mobile-team` on chat.zulip.org and we'll be glad to discuss.

Reply:
I am having a hard time writing tests for the new implementation, as I cannot differentiate between a header being sticky or normal. So I will start a thread in `#mobile-team`.

Reply:
With the new implementation, I am having a hard time writing tests for this feature. I am not even sure what to test.
In the previous implementation with the `BuildContext` only passed to the sticky headers, I could check for the value of the `sectionContext` for nullability, so a sticky header before collapsing would have a non-null `context` and after collapsing I would check for the presence of the same header with a non-null value. The same for a normal header having a null `context` and present on the screen. Now the `sectionContext` is a non-null value all the time, so it's hard to differentiate between the two states (types) of headers.

Or as you said before, should I share this on a thread in `#mobile-team`?!

Reply:
Yeah, a chat thread would be the best medium for this.